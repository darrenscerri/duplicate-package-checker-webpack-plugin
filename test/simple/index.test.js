var webpack = require("webpack");
var assert = require("assert");
var MakeConfig = require("./make.webpack.config");

describe("Simple dependency tree", function() {
  it("should output warnings", function(done) {
    webpack(MakeConfig(), function(err, stats) {
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      expect(stats.compilation.warnings[1].message).toMatchSnapshot();
      assert(stats.compilation.warnings.length === 2);
      done();
    });
  });

  it("should output warnings in verbose", function(done) {
    webpack(MakeConfig({ verbose: true }), function(err, stats) {
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      expect(stats.compilation.warnings[1].message).toMatchSnapshot();
      assert(stats.compilation.warnings.length === 2);
      done();
    });
  });

  it("should output errors", function(done) {
    webpack(MakeConfig({ emitError: true }), function(err, stats) {
      expect(stats.compilation.errors[0].message).toMatchSnapshot();
      expect(stats.compilation.errors[1].message).toMatchSnapshot();
      assert(stats.compilation.errors.length === 2);
      assert(stats.compilation.warnings.length === 0);
      done();
    });
  });

  it("should ignore excluded duplicates by name", function(done) {
    webpack(
      MakeConfig({
        exclude: function(instance) {
          return instance.name === "a";
        }
      }),
      function(err, stats) {
        expect(stats.compilation.warnings[0].message).toMatchSnapshot();
        assert(stats.compilation.warnings.length === 1);
        done();
      }
    );
  });

  it("should ignore excluded duplicates by issuer", function(done) {
    webpack(
      MakeConfig({
        exclude: function(instance) {
          return instance.issuer === "./entry.js";
        }
      }),
      function(err, stats) {
        assert(stats.compilation.warnings.length === 0);
        done();
      }
    );
  });
  it("should respect showHelp option", function(done) {
    webpack(
      MakeConfig({
        showHelp: false
      }),
      function(err, stats) {
        expect(stats.compilation.warnings[0]).toMatchSnapshot();
        expect(stats.compilation.warnings[1]).toMatchSnapshot();
        done();
      }
    );
  });
});
