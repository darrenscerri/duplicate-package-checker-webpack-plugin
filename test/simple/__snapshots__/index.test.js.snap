// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple dependency tree should ignore excluded duplicates by name 1`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/d/~/b[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should output errors 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
"
`;

exports[`Simple dependency tree should output errors 2`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/d/~/b[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should output warnings 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
"
`;

exports[`Simple dependency tree should output warnings 2`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/d/~/b[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should output warnings in verbose 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m from [37m[1m./entry.js[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m from [37m[1m./~/b/index.js[22m[39m
"
`;

exports[`Simple dependency tree should output warnings in verbose 2`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m from [37m[1m./entry.js[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/d/~/b[22m[39m from [37m[1m./~/c/~/d/index.js[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should respect showHelp option 1`] = `
[Error: a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
]
`;

exports[`Simple dependency tree should respect showHelp option 2`] = `
[Error: b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/d/~/b[22m[39m
]
`;
